### PR TITLE
add authentication status to PIResponse

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**

### CODE REVIEW

File: src/main/java/org/privacyidea/PIResponse.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
     public String jsonRPCVersion = "";
     public boolean status = false;
     public boolean value = false;
+    public String authentication = "";
```
Comment (nilsbehlen) [inline src/main/java/org/privacyidea/PIResponse.java:42]: this should be of type AuthenticationStatus, not string

File: src/main/java/org/privacyidea/AuthenticationStatus.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.privacyidea;
+
+public enum AuthenticationStatus
+{
+    CHALLENGE,
```
Comment (nilsbehlen) [inline src/main/java/org/privacyidea/AuthenticationStatus.java:5]: Probably need a default case (like "NONE") for initalization

File: src/main/java/org/privacyidea/PIResponse.java (hunk 1 above)
Comment (nilsbehlen) [reply src/main/java/org/privacyidea/PIResponse.java:42]: And should be initialized to the default case (like "NONE")

**Review by nilsbehlen — CHANGES_REQUESTED** [review]

File: src/main/java/org/privacyidea/PIResponse.java (hunk 1 above)
Comment (lukasmatusiewicz) [reply src/main/java/org/privacyidea/PIResponse.java]: fixed

File: src/main/java/org/privacyidea/AuthenticationStatus.java (hunk 2 above)
Comment (lukasmatusiewicz) [reply src/main/java/org/privacyidea/AuthenticationStatus.java:5]: fixed

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**